changer-d-epoque.csl

```diff
@@ -243,8 +243,8 @@
       </group>
       <text variable="container-title" font-style="italic"/>
       <text macro="collection" prefix=" "/>
-      <text macro="publisher" prefix=", " suffix=", "/>
-      <text macro="year-date" suffix=", "/>
+      <text macro="publisher" prefix=", "/>
+      <text macro="year-date" prefix=", " suffix=", "/>
       <text macro="pages"/>
```

Comment:
Ideally we'd rewrite the entire block to put all the delimiting punctuation on groups, but you could already make a small start by changing:

```xml
      <text macro="collection" prefix=" "/>
      <text macro="publisher" prefix=", "/>
      <text macro="year-date" prefix=", " suffix=", "/>
      <text macro="pages"/>
```

to

```xml
      <group prefix=" " delimiter=", "/>
        <text macro="collection"/>
        <text macro="publisher"/>
        <text macro="year-date"/>
        <text macro="pages"/>
      </group>
```

That should make things more robust in case one of the macros doesn't produce any output.